apps/server/src/endpoints.ts

```diff
@@ -86,10 +88,20 @@ export const createGame =
       switch (body.modelType) {
         case ModelType.THREAT_DRAGON: {
           // TODO: validation
-          await gameServer.db.setModel(
-            matchID,
-            JSON.parse(body.model as string) as ThreatDragonModel,
-          );
+          logEvent('vishal printing body');
+          logEvent(`printing body: ${body.model}`);
+          var model = JSON.parse(body.model as string) as ThreatDragonModel;
+          await gameServer.db.setModel(matchID,model);
+          break;
+        }
+
+        case ModelType.THREAT_DRAGON_V2: {
+          // TODO: validation
```

Comment:
Does Threat Dragon provide a lib for validation?

Reply:
There is a json schema. We could do validation based on that. Personally, I would prefer to convert that to a zod schema, though.